controllers/opentelemetrycollector_controller.go

```diff
@@ -152,25 +126,34 @@ func (r *OpenTelemetryCollectorReconciler) findClusterRoleObjects(ctx context.Co
 	return ownedObjects, nil
 }
 
-// getConfigMapsToRemove returns a list of ConfigMaps to remove based on the number of ConfigMaps to keep.
-// It keeps the newest ConfigMap, the `configVersionsToKeep` next newest ConfigMaps, and returns the remainder.
-func (r *OpenTelemetryCollectorReconciler) getConfigMapsToRemove(configVersionsToKeep int, configMapList *corev1.ConfigMapList) []corev1.ConfigMap {
+// removeKeptConfigMaps removes old ConfigMaps that we want to keep from the map of owned objects.
+// Normally the controller would delete them after determining they're not in the list of desired objects generated
+// from the OpenTelemetryCollector CR, but we want to keep them around.
+func removeKeptConfigMapVersions(configVersionsToKeep int, ownedObjects map[types.UID]client.Object) {
```

Comment:
I think the best way would be to change the function to return the CMs that should be kept, then it could be named `getConfigMapsToKeep`, it would take a slice of configmaps and return one. Then the calling function could build the slice from the map, call the function, and delete the output from the map.

Reply:
Refactored it to a very simple form and added some unit tests.